clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/AbstractEnableDisableOpenstackAtomicOperation.groovy

```diff
@@ -0,0 +1,76 @@
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package com.netflix.spinnaker.clouddriver.openstack.deploy.ops.servergroup
+
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
+import com.netflix.spinnaker.clouddriver.openstack.deploy.description.servergroup.OpenstackServerGroupAtomicOperationDescription
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
+import org.openstack4j.model.network.ext.LbPool
+
+abstract class AbstractEnableDisableOpenstackAtomicOperation implements AtomicOperation<Void> {
+  abstract boolean isDisable()
+
+  abstract String getPhaseName()
+
+  OpenstackServerGroupAtomicOperationDescription description
+
+  AbstractEnableDisableOpenstackAtomicOperation(OpenstackServerGroupAtomicOperationDescription description) {
+    this.description = description
+  }
+
+  protected static Task getTask() {
+    TaskRepository.threadLocalTask.get()
+  }
+
+  @Override
+  Void operate(List priorOutputs) {
+    String verb = disable ? 'disable' : 'enable'
+    String presentParticipling = disable ? 'Disabling' : 'Enabling'
+
+    task.updateStatus phaseName, "Initializing $verb server group operation for $description.serverGroupName in " +
+                                 "$description.region..."
+
+    def instanceIds = description.credentials.provider.getInstanceIdsForStack(description.region, description.serverGroupName)
+
+    def poolsForRegion = description.credentials.provider.getAllLoadBalancerPools(description.region)
+
+    if (disable) {
+      task.updateStatus phaseName, "Deregistering instances from load balancers..."
+
+      for (String id : instanceIds) {
+        String ip = description.credentials.provider.getIpForInstance(description.region, id)
+        for (LbPool pool : poolsForRegion) {
+          String memberId = description.credentials.provider.getMemberIdForInstance(description.region, ip, pool)
+          description.credentials.provider.removeMemberFromLoadBalancerPool(description.region, memberId)
+        }
+      }
+    } else {
+      task.updateStatus phaseName, "Registering instances with load balancers..."
+
+      for (String id : instanceIds) {
+        String ip = description.credentials.provider.getIpForInstance(description.region, id)
+        for (LbPool pool : poolsForRegion) {
+          int port = description.credentials.provider.getInternalLoadBalancerPort(pool)
+          description.credentials.provider.addMemberToLoadBalancerPool(description.region, ip, pool.id, port, 1)
```

Comment:
1 is good. If you have other things to change, maybe make it a static field with a decent name so it is obvious.